### DISCUSSION TITLE
cpp/object_oriented/overloading_overriding_and_hiding

### DISCUSSION THREAD

**Discussion by giscus** [imarvinle/CSGuide#70] [Announcements]
# cpp/object_oriented/overloading_overriding_and_hiding

编程指北: 计算机系统学习指南, 操作系统, 计算机网络, C++, Java, 算法, 数据结构

https://csguide.cn/cpp/object_oriented/overloading_overriding_and_hiding.html

**Comment by ghost** [comment]
重写：必须是虚函数，参数列表相同，多态性。
隐藏：名字相同，参数列表不同或基类函数非虚函数，基类函数被隐藏。
重载：同类中名字相同，参数列表不同，编译器根据参数选择合适的函数。

**Comment by Abigail1985** [comment]
如果重写了虚函数，调用的是派生类的实现，即使是通过 Base*。
如果没有重写，调用的是基类的实现，因为派生类没有提供新行为。
